test/integration_test.go

```diff
@@ -3556,6 +3555,137 @@ func (ts *IntegrationTestSuite) TestUpdateSettingHandlerInHandler() {
+	run1, err := ts.client.ExecuteWorkflow(ctx, startOptions, ts.workflows.UpdateEntityWorkflow)
+	ts.NoError(err)
+	ts.NotNil(run1)
+
+	updateHandle, err := updateOp.Get(run1)
+	ts.NoError(err)
+	ts.ErrorContains(updateHandle.Get(ctx, &updateResult), "addend must be non-negative")
+
+	// TODO: test retry on update failure
+
+	// request fails because workflow is already running, does not send update
+	updateOp, err = client.PrepareUpdateWorkflowOperation(
+		client.UpdateWorkflowOptions{
+			UpdateName:   "update",
+			Args:         []any{1},
+			WaitForStage: client.WorkflowUpdateStageCompleted,
+		})
+	ts.NoError(err)
+
+	startOptions.WorkflowOperation = updateOp
+	// NOTE that WorkflowExecutionErrorWhenAlreadyStarted (defaults to false) has no impact
+	_, err = ts.client.ExecuteWorkflow(ctx, startOptions, ts.workflows.UpdateEntityWorkflow)
+	ts.ErrorContains(err, "Workflow execution is already running")
+
+	// request uses existing workflow, sends update and waits for completion
+	updateOp, err = client.PrepareUpdateWorkflowOperation(
+		client.UpdateWorkflowOptions{
+			UpdateName:   "update",
+			Args:         []any{1},
+			WaitForStage: client.WorkflowUpdateStageCompleted,
+		})
+	ts.NoError(err)
+
+	startOptions.WorkflowOperation = updateOp
+	startOptions.WorkflowIDConflictPolicy = enumspb.WORKFLOW_ID_CONFLICT_POLICY_USE_EXISTING
+	run2, err := ts.client.ExecuteWorkflow(ctx, startOptions, ts.workflows.UpdateEntityWorkflow)
+	ts.NoError(err)
+	ts.Equal(run1.GetRunID(), run2.GetRunID())
+
+	updateHandle, err = updateOp.Get(run2)
+	ts.NoError(err)
+	ts.NoError(updateHandle.Get(ctx, &updateResult))
+	ts.Equal(1, updateResult)
+
+	// looking up update result with invalid run fails
+	updateHandle, err = updateOp.Get(nil)
+	ts.ErrorContains(err, "invalid workflow run: is nil")
+	updateHandle, err = updateOp.Get(run1)
+	ts.ErrorContains(err, "invalid workflow run: does not belong to update operation")
+
+	// re-using same update operation fails
+	startOptions.WorkflowOperation = updateOp
+	_, err = ts.client.ExecuteWorkflow(ctx, startOptions, ts.workflows.UpdateEntityWorkflow)
+	ts.ErrorContains(err, "invalid WorkflowOperation: update operation was already executed")
+
+	// request uses existing workflow, sends update and waits for acceptance, polls for result, completes workflow
+	updateOp, err = client.PrepareUpdateWorkflowOperation(
+		client.UpdateWorkflowOptions{
+			UpdateName:   "update",
+			Args:         []any{1},
+			WaitForStage: client.WorkflowUpdateStageAccepted,
+		})
+	ts.NoError(err)
+
+	startOptions.WorkflowOperation = updateOp
+	run4, err := ts.client.ExecuteWorkflow(ctx, startOptions, ts.workflows.UpdateEntityWorkflow)
+	ts.NoError(err)
+	ts.Equal(run1.GetRunID(), run4.GetRunID())
+
+	updateHandle, err = updateOp.Get(run4)
+	ts.NoError(err)
+	ts.NoError(updateHandle.Get(ctx, &updateResult))
+	ts.Equal(2, updateResult)
+
+	ts.NoError(run1.Get(ctx, &workflowResult))
+	ts.Equal(2, workflowResult)
+}
```

Comment:
NOTE: I wanted to break out the individual test cases into nested tests via `ts.Run` but none of the other tests are doing that. I think it would help with scoping (it's easy to accidentally use the wrong variable) and readability. Let me know what you think.